### PR TITLE
getting publisher working with different types (json only)

### PR DESCRIPTION
**Pull request by stevegal**
Note a PR in core is also required for this to work.

Adds support in jenkins test publisher to support more formats than just cucumber

### CODE REVIEW

**Comment by stevegal** [comment]
Note this requires an update in the core in order to build. Not sure how these types of changes can work now things have split.

Also, The UI build was failing locally with a missing dependency and I had to manually do an npm install in that project.

**Comment by stevegal** [comment]
requires https://github.com/Hygieia/hygieia-core/pull/56

File: hygieia-jenkins-plugin/src/test/java/hygieia/transformer/MochaSpecToTestCapabilityTransformerTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.capitalone.dashboard.model.quality.MochaJsSpecReport;
+import com.capitalone.dashboard.request.BuildDataCreateRequest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
```
Comment (Sbrenthughes) [inline hygieia-jenkins-plugin/src/test/java/hygieia/transformer/MochaSpecToTestCapabilityTransformerTest.java]: Can you remove the * import here?

File: hygieia-jenkins-plugin/src/test/java/hygieia/transformer/TestResultVisitorTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.capitalone.dashboard.model.quality.MochaJsSpecReport;
+import com.capitalone.dashboard.request.BuildDataCreateRequest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
```
Comment (Sbrenthughes) [inline hygieia-jenkins-plugin/src/test/java/hygieia/transformer/TestResultVisitorTest.java]: Avoid * imports.

File: hygieia-jenkins-plugin/src/test/java/jenkins/plugins/hygieia/workflow/HygieiaTestPublishStepTest.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package jenkins.plugins.hygieia.workflow;
+
+import static org.junit.Assert.*;
```
Comment (Sbrenthughes) [inline hygieia-jenkins-plugin/src/test/java/jenkins/plugins/hygieia/workflow/HygieiaTestPublishStepTest.java]: Avoid * imports.

File: hygieia-jenkins-plugin/src/test/java/hygieia/transformer/CucumberJsonToTestResultTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import com.capitalone.dashboard.model.quality.CucumberJsonReport;
+import com.capitalone.dashboard.request.BuildDataCreateRequest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
```
Comment (Sbrenthughes) [inline hygieia-jenkins-plugin/src/test/java/hygieia/transformer/CucumberJsonToTestResultTransformerTest.java]: Avoid * imports

File: hygieia-jenkins-plugin/src/test/java/hygieia/transformer/CodeQualityVisiteeDeserializerTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import com.capitalone.dashboard.model.quality.QualityVisitee;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.junit.Assert.*;
```
Comment (Sbrenthughes) [inline hygieia-jenkins-plugin/src/test/java/hygieia/transformer/CodeQualityVisiteeDeserializerTest.java]: Avoid * imports

**Comment by Sbrenthughes** [comment]
@rvema can you do a quick review here?

File: collectors/build/jenkins-codequality/pom.xml [hunk 6]
```diff
@@ -83,7 +83,7 @@
         <dependency>
             <groupId>com.capitalone.dashboard</groupId>
             <artifactId>core</artifactId>
-            <version>3.0.1</version>
+            <version>3.1.0-SNAPSHOT</version>
```
Comment (rvema) [inline collectors/build/jenkins-codequality/pom.xml]: Can you update it to 3.1.1-SNAPSHOT

File: hygieia-jenkins-plugin/pom.xml [hunk 7]
```diff
@@ -121,7 +121,7 @@
         <dependency>
             <groupId>com.capitalone.dashboard</groupId>
             <artifactId>core</artifactId>
-            <version>LATEST</version>
+            <version>3.1.0-SNAPSHOT</version>
```
Comment (rvema) [inline hygieia-jenkins-plugin/pom.xml]: please update the revision to 3.1.1-SNAPSHOT

**Comment by stevegal** [comment]
No jelly file changes required as it discovers the type from the given input

**Comment by rvema** [comment]
@stevegal can you update the version of hygieia-core? the build for Hygieia needs to be fixed

**Pull request merged**